### PR TITLE
Add end to end tests

### PR DESCRIPTION
**Pull request by Christdej**

### CODE REVIEW

File: backend/api/api.csproj [hunk 1]
```diff
@@ -27,4 +27,8 @@
     <PackageReference Include="MQTTnet.Extensions.ManagedClient" Version="4.0.2.221" />
   </ItemGroup>
 
+  <ItemGroup>
+     <InternalsVisibleTo Include="api" />
```
Comment (GodVenn) [inline backend/api/api.csproj:31]: What is this for?

File: backend/api.test/ComponentTest.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            _client.Dispose();
+            GC.SuppressFinalize(this);
+        }
+
+        [Fact]
+        public async Task MissionsTest()
+        {
+            string url = "/missions";
+            var response = await _client.GetAsync(url);
+            var msg = await response.Content.ReadFromJsonAsync<List<Mission>>(
```
Comment (GodVenn) [inline backend/api.test/ComponentTest.cs:79]:
```suggestion
            var missions = await response.Content.ReadFromJsonAsync<List<Mission>>(
```

File: backend/api.test/ComponentTest.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.True(msg != null && msg.Count == 3);
+        }
+
+        [Fact]
+        public async Task RobotsTest()
+        {
+            string url = "/robots";
+            var response = await _client.GetAsync(url);
+            var msg = await response.Content.ReadFromJsonAsync<List<Robot>>(
```
Comment (GodVenn) [inline backend/api.test/ComponentTest.cs:93]:
```suggestion
            var robots = await response.Content.ReadFromJsonAsync<List<Robot>>(
```

File: backend/api.test/DbContextTestSetup.cs [hunk 4]
```diff
@@ -54,4 +64,104 @@ public class DatabaseCollection : ICollectionFixture<DatabaseFixture>
...
+            await Task.Run(() => Thread.Sleep(1));
+            return new HttpResponseMessage();
+        }
+
+        public async Task<IsarMissionDefinition> GetIsarMissionDefinition(EchoMission echoMission)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new IsarMissionDefinition(new List<IsarTaskDefinition>());
+        }
+    }
```
Comment (GodVenn) [inline backend/api.test/DbContextTestSetup.cs:166]: These classes should be in their own files, maybe in a new "mocks" folder

File: backend/api/Database/Context/InitDb.cs [hunk 5]
```diff
@@ -116,6 +116,11 @@ private static List<Mission> GetMissions()
 
     public static void PopulateDb(FlotillaDbContext context)
     {
+        // To make sure we are not trying to initialize database more than once during tests
```
Comment (GodVenn) [inline backend/api/Database/Context/InitDb.cs:119]: I thought we had solved this

**Review by GodVenn — CHANGES_REQUESTED** [review]
Very cool that it works!

File: backend/api/api.csproj (hunk 1 above)
Comment (Christdej) [reply backend/api/api.csproj:31]: To make the api internally visible to the test project
https://learn.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0#basic-tests-with-the-default-webapplicationfactory-1

File: backend/api.test/xunit.runner.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "$schema": "https://xunit.net/schema/current/xunit.runner.schema.json",
+  "parallelizeTestCollections": false
```
[the remark this answers is not in the document]
Comment (Christdej) [reply backend/api.test/xunit.runner.json:3]: This should be removed, solved problem using something else

File: backend/api.test/DbContextTestSetup.cs (hunk 4 above)
Comment (Christdej) [reply backend/api.test/DbContextTestSetup.cs:166]: Agreed

File: backend/api.test/ComponentTest.cs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                new JsonSerializerOptions
+                {
+                    Converters = { new JsonStringEnumConverter() },
+                });
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.True(msg != null && msg.Count == 3);
```
[the remark this answers is not in the document]
Comment (Christdej) [reply backend/api.test/ComponentTest.cs:99]: Agreed

File: backend/api.test/ComponentTest.cs (hunk 3 above)
Comment (Christdej) [reply backend/api.test/ComponentTest.cs:93]: Agreed

File: backend/api.test/ComponentTest.cs [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        {
+            string url = "/missions";
+            var response = await _client.GetAsync(url);
+            var msg = await response.Content.ReadFromJsonAsync<List<Mission>>(
+                new JsonSerializerOptions
+                {
+                    Converters = { new JsonStringEnumConverter() },
+                });
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.True(msg != null && msg.Count == 3);
```
[the remark this answers is not in the document]
Comment (Christdej) [reply backend/api.test/ComponentTest.cs:85]: Agreed

File: backend/api.test/ComponentTest.cs (hunk 2 above)
Comment (Christdej) [reply backend/api.test/ComponentTest.cs:79]: Agreed

File: backend/api/Database/Context/InitDb.cs (hunk 5 above)
Comment (Christdej) [reply backend/api/Database/Context/InitDb.cs:119]: So though I, but Xunit insists on initiating the database twice

**Review by GodVenn — COMMENTED** [review]
Should we rename ComponentTest.cs to ´EndpointTests.cs´ ?

**Review by GodVenn — APPROVED** [review]
LGTM.
Great work!

**Pull request merged**